### PR TITLE
Add docs for NIA Terraform driver

### PR DESCRIPTION
**Pull request by findkim**
First portion of docs covering the Terraform driver and generated Terraform configuration files. Will make a follow-up PR to finish the remaining sections: Compatible Terraform Modules, Module Input Variables.

[Docs preview](https://deploy-preview-8871--consul-docs-preview.netlify.app/docs/nia/network-drivers)

### CODE REVIEW

File: website/pages/docs/nia/network-drivers.mdx [hunk 1]
```diff
@@ -6,10 +6,49 @@ description: >-
...
+
+~> **Note:** Although Terraform state files for task workspaces are independent, this does not guarantee the infrastructure changes from concurrent task executions are independent. Ensure that modules across all tasks are not modifying the same resource objects or have overlapping changes that may result in race conditions during automation.
+
+Within the Consul-Terraform-Sync configuration for a task, practitioners can select the desired module to run for the task as well the set of services for the module to execute on. Each task executed by the Terraform driver corresponds to an automated root module that calls the selected module in an isolated Terraform environment. Consul-Terraform-Sync will manage concurrent execution of these tasks.
+
+### Root Module
+
+The root module is simple in structure and proxies Consul service information, configuration, and other variables to the module. The content of the files that make up the root module are sourced from Consul Terraform Sync configuration, information for task's module to use as the automation playbook, and the Consul Catalog for discovering service information.
```
Comment (lornasong) [inline website/pages/docs/nia/network-drivers.mdx:27]: we haven't really picked with/without dashes. it seems like you're mostly with dashes. maybe "Consul Terraform Sync" => "Consul-Terraform-Sync"?

File: website/pages/docs/nia/network-drivers.mdx [hunk 2]
```diff
@@ -6,10 +6,49 @@ description: >-
...
+Autogenerated root modules for tasks are stored in local subdirectories of the Terraform working directory. By default, the working directory `sync-tasks` is created in the current directory. To configure where Terraform configuration files are stored, set [`working_dir`](/docs/nia/installation/configuration#working_dir) for the Terraform driver to the desired path.
+
+```
+sync-tasks/
+  my-task/
+    main.tf
+    variables.tf
+    terraform.tfvars
+    terraform.tfvars.tmpl
+```
```
Comment (lornasong) [inline website/pages/docs/nia/network-drivers.mdx:38]: ++ I found this overview really helpful

File: website/pages/docs/nia/network-drivers.mdx [hunk 3]
```diff
@@ -6,10 +6,49 @@ description: >-
...
+
+The following files of the root module are generated for each task. An [example of a root module created by Consul-Terraform-Sync](https://github.com/hashicorp/consul-terraform-sync/tree/master/examples) can be found in the project repository.
+* `main.tf` - The main file contains the terraform block, provider blocks, and a module block calling the module configured for the task.
+    * `terraform` block - The corresponding provider source and versions for the task from the configuration files are placed into this block for the root module. The Terraform backend from the configuration is also templated here.
+    * `provider` blocks - The provider blocks generated in the root module resemble the provider blocks in the configuration. They have identical arguments present nad are set from the intermediate variable created per provider.
```
Comment (lornasong) [inline website/pages/docs/nia/network-drivers.mdx:43]: suggestion to change: "nad" => "and"

File: website/pages/docs/nia/network-drivers.mdx [hunk 4]
```diff
@@ -6,10 +6,49 @@ description: >-
...
+    * `module` block - The module block is where the task's module is called as a [child module](https://www.terraform.io/docs/configuration/modules.html#calling-a-child-module). Required and optional input variables are passed as arguments to the module.
+* `variables.tf` - This file contains 2 types of variable defintions. The required `services` input variable which determines module compatibility with Consul-Terraform Sync (read more on [compatible Terraform modules](#compatible-terraform-modules) for more details) and various intermediate variables used to dynamically configure providers. These intermediate provider variables are interpolated from the provider blocks and arguments configured in the Consul-Terraform-Sync configuration.
```
Comment (lornasong) [inline website/pages/docs/nia/network-drivers.mdx:45]: suggestion to change: "defintions" => "definitions"

File: website/pages/docs/nia/network-drivers.mdx [hunk 5]
```diff
@@ -6,10 +6,49 @@ description: >-
...
+* `variables.module.tf` - This file is conditionally created if there are [variables configured for the task](/docs/nia/installation/configuration#variable_files) and contains the interpolated variable declarations that match the variables from configuration. These are then used to proxy the configured variables to the module through explicit assignment in the module block.
+* `terraform.tfvars` - The variable definitions file is where the services input variable is assigned values from the Consul Catalog. It is periodically updated to reflect the current state of the configured set of services for the task.
```
Comment (lornasong) [inline website/pages/docs/nia/network-drivers.mdx:47]: Just want to make sure I'm on the right page, it seems like both `.tf` and `.tfvars` files are called variable definitions files. is that right?

**Review by lornasong — APPROVED** [review]
looks good to me. left a few super minor things.

you had previously drawn up a diagram of how the files all relate to each other. it was really helpful for me to understand how everything fits together. super off-the-cuff idea to maybe include it or a version of it in the docs?!

File: website/pages/docs/nia/network-drivers.mdx (hunk 1 above)
Comment (findkim) [reply website/pages/docs/nia/network-drivers.mdx:27]: I have been manually changing these occurrences, missed this, thank you!

File: website/pages/docs/nia/network-drivers.mdx (hunk 4 above)
Comment (findkim) [reply website/pages/docs/nia/network-drivers.mdx:45]: 🙇

File: website/pages/docs/nia/network-drivers.mdx (hunk 3 above)
Comment (findkim) [reply website/pages/docs/nia/network-drivers.mdx:43]: 🙇‍♂️

File: website/pages/docs/nia/network-drivers.mdx (hunk 5 above)
Comment (findkim) [reply website/pages/docs/nia/network-drivers.mdx:47]: To be honest I'm not confident and I've been heavily relying on the Terraform docs. `.tfvars` is variable definitions files ([source](https://www.terraform.io/docs/configuration/variables.html#variable-definitions-tfvars-files)), and from my understanding the `variables.tf` is just convention to centrally locate variable declarations.

It looks like I misused "definitions" in the `variables.tf` section, updated to "This file contains 2 types of variable declarations".

**Comment by findkim** [comment]
Appreciate the review @lornasong! I'll skip on the diagram, and if later down the line we find that we need it I can ask about getting a more official looking one.

**Pull request merged**